pkg/cachemanager/cachemanager.go

```diff
@@ -160,63 +157,57 @@ func (c *CacheManager) replaceWatchSet(ctx context.Context) error {
 	newWatchSet.Add(c.gvksToSync.GVKs()...)
 	c.gvksToDeleteFromCache.AddSet(c.watchedSet.Difference(newWatchSet))
 
-	var innerError error
-	c.watchedSet.Replace(newWatchSet, func() {
-		// *Note the following steps are not transactional with respect to admission control
+	if c.danglingWatches || !c.watchedSet.Equals(newWatchSet) {
```

Comment:
We don't want to gate `replaceWatchSet` like we're doing. I think this same (or similar) idea had shown up earlier. We shouldn't do it because it may short-circuit retries due to failure adding watches.

The if statement makes sense, but I would run it in the `manageCache` goroutine only.